src/common/vector_operations/vector_copy.cpp

```diff
@@ -212,23 +212,18 @@ void VectorOperations::Copy(const Vector &source, Vector &target, idx_t source_c
 		VectorOperations::Copy(child, target, dict_sel, source_count, source_offset, target_offset);
 		break;
 	}
-	case VectorType::CONSTANT_VECTOR:
-		VectorOperations::Copy(source, target, ConstantVector::ZERO_SELECTION_VECTOR, source_count, source_offset,
-		                       target_offset);
+	case VectorType::CONSTANT_VECTOR: {
+		SelectionVector owned_sel;
+		auto sel = ConstantVector::ZeroSelectionVector(source_count, owned_sel);
```

Comment:
Not a show stopper, but maybe only do this if `source_count > STANDARD_VECTOR_SIZE`?

Reply:
That actually happens inside the `ConstantVector::ZeroSelectionVector`!

```cpp
const SelectionVector *ConstantVector::ZeroSelectionVector(idx_t count, SelectionVector &owned_sel) {
	if (count <= STANDARD_VECTOR_SIZE) {
		return &ConstantVector::ZERO_SELECTION_VECTOR;
	}
	owned_sel.Initialize(count);
	for (idx_t i = 0; i < count; i++) {
		owned_sel.set_index(i, 0);
	}
	return &owned_sel;
}
```